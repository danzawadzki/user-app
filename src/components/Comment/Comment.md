Default comment with all props:

```jsx
<Comment
	author={{
		name: 'Random User',
		avatar: {
			small: 'https://randomuser.me/api/portraits/thumb/men/65.jpg',
			regular: 'https://randomuser.me/api/portraits/med/men/65.jpg',
			large: 'https://randomuser.me/api/portraits/men/65.jpg'
		}
	}}
	comment="Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula."
	timestamp="10/11/2018"
/>
```

Comment written by author with out avatar:

```jsx
<Comment
	author={{
		name: 'Random User'
	}}
	comment="Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula."
	timestamp="10-11-2018"
/>
```
