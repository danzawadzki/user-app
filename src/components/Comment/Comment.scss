@import '../../assets/styles/colors';

.Comment {
	&__header {
		display: flex;
		align-items: center;

		//Comment Header block
		& > .header {
			&__avatar {
				width: 5rem;
				flex-grow: 0;
			}

			&__name {
				color: $blue__navy;
				flex-grow: 1;
			}

			&__date {
				color: $blue--light;
				flex-grow: 0;
			}
		}
	}

	&__content {
		padding-left: 50px;
	}
}
