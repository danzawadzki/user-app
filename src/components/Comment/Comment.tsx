import * as React from 'react';
import './Comment.scss';
import Avatar from '../Avatar/Avatar';
import { timeAgo } from '../../utilities/helpers/date.helpers';

export interface ICommentAuthor {
	/** Object containing avatar urls. */
	avatar?: {
		/** Optional image url for small screen devices (screen < 600px) */
		small?: string;
		/** Required image ulr for medium screen devices (600px =< dev <= 1600px) or all else if not provided. */
		regular?: string;
		/** Optional image url for large screen devices (1900px < screen) */
		large?: string;
	};
	name: string;
}

export interface IComment {
	/** Comment author object */
	author: ICommentAuthor;
	/** Content of the comment.  */
	comment: string;
	/** Comment timestamp */
	timestamp: string;
}

/**
 * Comment component.
 *
 * @version 1.0.0
 */
const Comment = ({ author: { avatar, name }, comment, timestamp }: IComment) => {
	return (
		<div className="Comment">
			<div className="Comment__header">
				<div className="header__avatar">
					<Avatar name={name} avatar={avatar} isSmaller={true} />
				</div>
				<div className="header__name">{name}</div>
				<div className="header__date">
					{timeAgo(new Date(timestamp), new Date())}
				</div>
			</div>
			<div className="Comment__content">{comment}</div>
		</div>
	);
};

export default Comment;
