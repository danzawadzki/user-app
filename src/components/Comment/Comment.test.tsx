import React from 'react';
import { shallow } from 'enzyme';
import Comment from './Comment';

describe('Comment component: ', () => {
	//Shallow rendered component
	const author = {
		name: 'Joe Black',
		avatars: {
			small: 'https://randomuser.me/api/portraits/thumb/men/65.jpg',
			regular: 'https://randomuser.me/api/portraits/med/men/65.jpg',
			large: 'https://randomuser.me/api/portraits/men/65.jpg'
		}
	};
	const comment =
		'Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula.';
	const timestamp = '10/11/2018';
	const component = shallow(
		<Comment author={author} comment={comment} timestamp={timestamp} />
	);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should render correctly with all props', () => {
		expect(component.find('.Comment')).toBeDefined();
		expect(component.find('.Comment__header').exists()).toBeTruthy();
		expect(component.find('.Comment__content').exists()).toBeTruthy();
	});

	it('should render correctly with all props', () => {
		expect(component.html()).toContain(author.name);
		expect(component.html()).toContain(comment);
	});
});
