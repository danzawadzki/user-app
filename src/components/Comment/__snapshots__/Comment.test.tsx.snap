// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component:  should render match snapshot 1`] = `
<div
  className="Comment"
>
  <div
    className="Comment__header"
  >
    <div
      className="header__avatar"
    >
      <Avatar
        isSmaller={true}
        name="Joe Black"
      />
    </div>
    <div
      className="header__name"
    >
      Joe Black
    </div>
    <div
      className="header__date"
    >
      1m
    </div>
  </div>
  <div
    className="Comment__content"
  >
    Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula.
  </div>
</div>
`;
